psamm/commands/randomsparse.py

```diff
@@ -91,173 +87,20 @@ def run(self):
-                    'Marking gene {} as essential because the objective'
-                    ' reaction depends on this gene...'.format(testing_gene))
-                essential.add(testing_gene)
-                continue
-
-            if len(deleted_reactions) == 0:
-                logger.info(
-                    'No reactions were removed when gene {}'
-                    ' was deleted'.format(testing_gene))
-                deleted.add(testing_gene)
-                gene_assoc = new_gene_assoc
-                continue
-
-            logger.info(
-                'Reactions removed when gene {} is deleted: {}'.format(
-                    testing_gene, len(deleted_reactions)))
-            logger.info('Deleted reactions: {}'.format(
-                ', '.join(deleted_reactions)))
-
-            constraints = []
-            for reaction in deleted_reactions:
-                flux_var = prob.get_flux_var(reaction)
-                c, = prob.prob.add_linear_constraints(flux_var == 0)
-                constraints.append(c)
-
-            try:
-                prob.maximize(obj_reaction)
-            except fluxanalysis.FluxBalanceError:
-                logger.info(
-                    'FBA is infeasible, marking {} as essential'.format(
-                        testing_gene))
-                for c in constraints:
-                    c.delete()
-                essential.add(testing_gene)
-                continue
-
-            logger.debug('Reaction {} has flux {}'.format(
-                obj_reaction, prob.get_flux(obj_reaction)))
-
-            if prob.get_flux(obj_reaction) < flux_threshold:
-                for c in constraints:
-                    c.delete()
-                essential.add(testing_gene)
-                logger.info('Gene {} was essential'.format(testing_gene))
-            else:
-                deleted.add(testing_gene)
-                reactions.difference_update(deleted_reactions)
-                gene_assoc = new_gene_assoc
-                logger.info('Gene {} was deleted'.format(testing_gene))
-
-        logger.info('Solving took {:.2f} seconds'.format(
-            time.time() - start_time))
-
-        for gene_id in sorted(genes):
-            value = 0 if gene_id in deleted else 1
-            print('{}\t{}'.format(gene_id, value))
+            strategy = randomsparse.GeneDeletionStrategy(
+                self._mm, randomsparse.get_gene_associations(self._model))
 
-        logger.info('Essential genes: {}/{}'.format(
-            len(essential), len(genes)))
+        essential, deleted = randomsparse.random_sparse(
+            strategy, p, reaction, flux_threshold)
 
-        logger.info('Reactions in minimal network: {}'.format(
-            ', '.join(sorted(reactions))))
+        logger.info('Essential entity: {}/{}'.format(
+            len(essential), len(strategy.get_total())))
+        logger.info('Deleted entity: {}/{}'.format(
+            len(deleted), len(strategy.get_total())))
```

Comment:
The output should probably be `Essential reactions` or `Essential genes` depending on which strategy was chosen.